x-pack/plugins/ingest_manager/server/services/epm/packages/install.ts

```diff
@@ -63,19 +63,41 @@ export async function installLatestPackage(options: {
   }
 }
 
+function isBulkInstallError(resp: BulkInstallResponse): resp is IBulkInstallPackageError {
+  return 'error' in resp && resp.error !== undefined;
+}
+
+async function getInstallationAndName(
+  savedObjectsClient: SavedObjectsClientContract,
+  pkgName: string
+) {
+  return {
+    pkgName,
+    installation: await getInstallation({ savedObjectsClient, pkgName }),
+  };
+}
+
+type GetInstallationReturnType = UnwrapPromise<ReturnType<typeof getInstallationAndName>>;
 export async function ensureInstalledDefaultPackages(
   savedObjectsClient: SavedObjectsClientContract,
   callCluster: CallESAsCurrentUser
-): Promise<Installation[]> {
+): Promise<GetInstallationReturnType[]> {
   const installations = [];
-  for (const pkgName in DefaultPackages) {
-    if (!DefaultPackages.hasOwnProperty(pkgName)) continue;
-    const installation = ensureInstalledPackage({
-      savedObjectsClient,
-      pkgName,
-      callCluster,
-    });
-    installations.push(installation);
+  const bulkResponse = await bulkInstallPackages({
+    savedObjectsClient,
+    packagesToUpgrade: Object.values(DefaultPackages),
+    callCluster,
+  });
+
+  for (const resp of bulkResponse) {
+    if (isBulkInstallError(resp)) {
+      throw new Error(resp.error);
+    } else {
+      // getInstallation can return undefined so we'll tack on the name so when we throw an error in setup we can
+      // reference the name of the package that we failed to retrieve the installation information for
+      const installation = getInstallationAndName(savedObjectsClient, resp.name);
```

Comment:
Would it be better to `await` the `getInstallation` call here? If we did then I wouldn't have to change the return value and would just check if the result was `undefined` and throw the error here instead of in `setup.ts`.